pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1810,6 +1810,28 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
         return closeFuture;
     }
 
+    private boolean isClosed() {
```

Comment:
It looks like the issue is due to ManagedLedger being closed, do these checks can make sure the ManagedLedger is open?

Reply:
Good point, but I am thinking about the following things
- The marker that is used to confirm whether a topic is closed or not is needed, I want to add a `State` for `PersistentTopic`, which will combine `isFenced`, `isClosingOrDeleted`...
- And replicator relates to `PersistentTopic`, not `ManagedLedger`, so using the state of Persistent Topic is better.

Reply:
Do we need to close the topic if the managedLedger closed?

Reply:
@poorbarcode is there a need to address this?

Reply:
> Do we need to close the topic if the managed ledger closed?

Yes, the managed ledger always be closed following the topic's closing. And regarding this case, the topic and managed ledger all be closed, but the retry task still runs 😂